### ISSUE TITLE
Cognitive Complexity of Protocols is too high

### ISSUE THREAD

**Issue by cfunk1210** [darpa-sail-on/sail-on-client#122]
The protocols are very hard to parse and find the important parts on.  It needs to be simplified (via subfunctions) to break down the complexity here.  

The goal is to break it down into semantic steps which are easier to digest with cognitive complexity of at most 30.

**Comment by as6520** [comment]
Closing since it s duplicate of https://github.com/darpa-sail-on/sail-on-client/issues/78
